src/main/java/reposense/git/GitUtil.java

```diff
@@ -80,14 +86,51 @@ static String convertToGitFormatsArgs(List<Format> formats) {
     /**
      * Returns the {@code String} command to specify the globs to exclude for `git log` command.
      */
-    static String convertToGitExcludeGlobArgs(List<String> ignoreGlobList) {
+    static String convertToGitExcludeGlobArgs(File root, List<String> ignoreGlobList) {
         StringBuilder gitExcludeGlobArgsBuilder = new StringBuilder();
         final String cmdFormat = " " + addQuote(":(exclude)%s");
         ignoreGlobList.stream()
-                .filter(item -> !item.isEmpty())
+                .filter(item -> !item.isEmpty() && pathExistsInRepo(root, item))
                 .map(ignoreGlob -> String.format(cmdFormat, ignoreGlob))
                 .forEach(gitExcludeGlobArgsBuilder::append);
 
         return gitExcludeGlobArgsBuilder.toString();
     }
+
+    /**
+     * Returns true if the {@code String} path is inside the current repository
+     */
+    static boolean pathExistsInRepo(File repoRoot, String path) {
+        String validPath = path;
+        if (path.contains("/*")) {
+            validPath = path.substring(0, path.indexOf(("/*")));
+        } else if (path.contains("*")) { // not in directories
+            return true;
+        }
+        return !childIsOutsideRepo(repoRoot, new File(repoRoot, validPath));
+    }
+
+    /**
+     * Returns true if the child path is outside repository folder
+     * @throws IOException if file system queries are needed
+     */
+    static boolean childIsOutsideRepo(File repoRoot, File child) {
+        try {
+            File canonicalRoot = repoRoot.getCanonicalFile();
+            File canonicalChild = child.getCanonicalFile();
+
+            File parentFile = canonicalRoot;
+            while (parentFile != null) {
+                if (canonicalChild.equals(parentFile)) {
+                    return true;
+                }
+                parentFile = parentFile.getParentFile();
```

Comment:
The logic here is a little bit weird. Why we are comparing canonicalRoot and its parents with canoicalChild?
Shouldn't we compare canonicalChild and its parents with canonicalRoot?

Reply:
The code here is trying to trace that the `child` does not exist outside the `root` folder:
![image](https://user-images.githubusercontent.com/30564248/54080132-101d1f00-431c-11e9-982c-5949e7f09078.png)

Reply:
Why a child outside the root must be a parent of root?

Reply:
For example, the parent root is A/B, the child root is A/C.

Wouldn't it be more straight forward to just check if the child is inside the root?

Reply:
i.e. The parent root is A/B/C and the child root is A/B/C/D/E/F, we just check if A/B/C/D/E/F or A/B/C/D/E or A/B/C/D or A/B/C equals the parent root A/B/C?

Reply:
> Why a child outside the root must be a parent of root?

That's to handle special situations and prevent runtime exception.

Reply:
> i.e. The parent root is A/B/C and the child root is A/B/C/D/E/F, we just check if A/B/C/D/E/F or A/B/C/D/E or A/B/C/D or A/B/C equals the parent root A/B/C?

The example is not valid for this situation.

Reply:
> > Why a child outside the root must be a parent of root?
> 
> That's to handle special situations and prevent runtime exception.

Do you mean that the child passed in the function is either a parent of the root or inside the root? Can you explain why is so?

Reply:
Yes.
Special situations like if users try to put folders existing outside of the git repository 
inside the ignore-glob-list, e.g. `../bin`, this will lead to `git log` creating a runtime exception.

Reply:
I have tested the globs mentioned in https://github.com/reposense/RepoSense/issues/398#issuecomment-470583555 and the problem is not solved. i.e. add `../nonExisting` will still result in runtime error.

Reply:
The problem is solved after I change the code to below:
```
    static boolean pathExistsInRepo(File repoRoot, String path) {
        String validPath = path;
        if (path.contains("/*")) {
            validPath = path.substring(0, path.indexOf(("/*")));
        } else if (path.contains("*")) { // not in directories
            return true;
        }
        return childIsInsideRepo(repoRoot, new File(repoRoot, validPath));
    }

    static boolean childIsInsideRepo(File repoRoot, File child) {
        try {
            File rootFile = repoRoot.getCanonicalFile();
            File childFile = child.getCanonicalFile();

            while (childFile != null) {
                if (childFile.equals(rootFile)) {
                    return true;
                }
                childFile = childFile.getParentFile();
            }
        } catch (IOException ex) {
            logger.log(Level.WARNING, "File system queries are needed.", ex);
            return false;
        } catch (SecurityException ex) {
            logger.log(Level.WARNING, "File cannot be accessed.", ex);
            return false;
        }
        return false;
    }
```

Reply:
> Yes.
> Special situations like if users try to put folders existing outside of the git repository
> inside the ignore-glob-list, e.g. `../bin`, this will lead to `git log` creating a runtime exception.

`../bin` will go the folder of with name `bin` in the same parent with the root but itself is not the parent of root.

You should try to add ignore glob list in author-csv to test.

Reply:
![image](https://user-images.githubusercontent.com/35621726/54080996-811e0000-4337-11e9-911a-d4fe9fd398b0.png)
This is the repo csv I am using.

Reply:
I get what you mean now. 
Didn't check specifically for `../bin` in the test case, but `../*java` passes the test.

Reply:
You may also add a `.exist()` to test whether the child exists. But not so sure if it is necessary. (Don't know how git glob works)